rcl/include/rcl/guard_condition.h

```diff
@@ -29,20 +29,22 @@ extern "C"
 #include "rcl/visibility_control.h"
 
 /// Internal rcl guard condition implementation struct.
-struct rcl_guard_condition_impl_t;
+struct rcl_guard_condition_impl_s;
+
+typedef struct rcl_guard_condition_impl_s rcl_guard_condition_impl_t;
```

Comment:
Similar question here; can we just drop this `typedef`?

Reply:
Same as above.